plugins/woocommerce/includes/class-wc-structured-data.php

```diff
@@ -577,4 +577,31 @@ public function generate_order_data( $order, $sent_to_admin = false, $plain_text
 
 		$this->set_data( apply_filters( 'woocommerce_structured_data_order', $markup, $sent_to_admin, $order ), true );
 	}
+
+	/**
+	 * Check if a GTIN is valid.
+	 * A GTIN is valid if:
+	 * Is a positive numeric string with a length of 8,12,13 or 14.
+	 *
+	 * @see https://schema.org/gtin
+	 * @param string $gtin The GTIN to check.
+	 * @return bool True if valid. False otherwise.
+	 */
+	public function is_valid_gtin( $gtin ) {
+		return is_string( $gtin ) && preg_match( '/^\d{8}$|^\d{12}$|^\d{13}$|^\d{14}$/', $gtin );
```

Comment:
Works the same, but would this be slightly less repetitious?

```suggestion
		return is_string( $gtin ) && preg_match( '/^(\d{8}|\d{12,14})$/', $gtin );
```

Or a [pattern like we use in G4W](https://github.com/woocommerce/google-listings-and-ads/blob/2.8.5/src/Product/WCProductAdapter.php#L816).